lib/rspec/core/configuration.rb

```diff
@@ -1613,6 +1613,9 @@ def extract_location(path)
           captures = match.captures
           path, lines = captures[0], captures[1][1..-1].split(":").map { |n| n.to_i }
           filter_manager.add_location path, lines
+        else
+          path, scoped_ids = path.split(/[\[\]]/)
```

Comment:
perhaps the regex needs a name? `split(match_square_brackets)` not convinced that's better, just throwing the idea out?